pkg/flow/worker/worker_pool.go

```diff
@@ -0,0 +1,124 @@
+package worker
+
+import (
+	"fmt"
+	"hash/fnv"
+	"math/rand"
+	"runtime"
+	"sync"
+)
+
+type Pool interface {
+	// Stop stops the worker pool. It does not wait to drain any internal queues. It must only be called once.
+	Stop()
+	// Submit submits a function to be executed by the worker pool on a random worker. Error is returned if the pool
+	// is unable to accept extra work.
+	Submit(func()) error
+	// SubmitWithKey submits a function to be executed by the worker pool, ensuring that only one
+	// job with given key can be queued at the time. Adding a job with a key that is already queued is a no-op.
+	// Error is returned if the pool is unable to accept extra work.
```

Comment:
What consequences does returning an error instead of blocking have? It seems the default queue size is 100. Is it possible that if we have too many components, this returns an error and the graph is not evaluated ok? Is the caller expected to retry in case of an error? It would be nice if the comment includes more information about this.